Add job to send matching data to UCAS

tijmenb opened the pull request:
##  Context

We're going to be sharing data with UCAS to find candidates who've applied to both services. 

## Changes proposed in this pull request

This is the first step in that process, it uploads the data every night to UCAS' Movit instance. It uses the Movit API, documented at https://transfer.ucasenvironments.com/swagger/ui.

Authentication works by fetching an access token using a username and password, configured in environment variables. Currently we have one email address that has to the folder we're uploading to - if you need access than you can look at the environment variables.

This uploads to a *test environment*. I'm not a 100% sure what will change between production and testing environments - definitely the username & password, likely the folder ID, perhaps the URL as well. We'll see once we have the production credentials. 

## Guidance to review

There's a follow up this to _download_ matched data from the servers. At that point we'll probably refactor this code a bit, because things will be shared (auth, folder IDs, etc).

## Link to Trello card

https://trello.com/c/vX5nXO2g/2331-arrange-matching-data-with-ucas

## Things to check

- [X] This code doesn't rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

duncanjbrown commented on app/services/ucas_matching/upload_matching_data.rb:
```diff
@@ -0,0 +1,35 @@
+    def perform
+      filename = "/tmp/dfe_apply_itt_applications_#{DateTime.now}.csv"
+
+      File.open(filename, 'a') do |f|
```
If this file is unique by timestamp, would `w` mode make more sense?

tijmenb replied:
Done!

duncanjbrown commented on spec/services/ucas_matching/upload_matching_data_to_ucas_spec.rb:
```diff
@@ -0,0 +1,20 @@
+
+      UCASMatching::UploadMatchingData.perform_async
+
+      expect(request).to have_been_made
```
Should we also check the authorization header here to make sure it's all hanging together?

tijmenb commented:
@duncanjbrown I've updated the PR. Replaced the unit test with a system test that also tests what happens either the API token or the upload fails. Added error handling.

duncanjbrown commented on app/services/ucas_matching/upload_matching_data.rb:
```diff
@@ -0,0 +1,51 @@
+        )
+
+      unless response.status.success?
+        raise ApiError, "HTTP #{response.status} when fetching access token: '#{response}'"
```
```suggestion
        raise ApiError, "HTTP #{response.status} when uploading to Movit: '#{response}'"
```
